flax/linen/module.py

```diff
@@ -1037,21 +1046,29 @@ def sow(self, col: str, name: str, value: T,
 
     Example::
 
+      import jax
+      import jax.numpy as jnp
+      import flax.linen as nn
+
       class Foo(nn.Module):
         @nn.compact
         def __call__(self, x):
           h = nn.Dense(4)(x)
           self.sow('intermediates', 'h', h)
           return nn.Dense(2)(h)
-      y, state = Foo.apply(params, x, mutable=['intermediates'])
+
+      x = jnp.ones(((16, 9)))
+      model = Foo()
+      variables = model.init(jax.random.PRNGKey(0), x)
```

Comment:
Why not the same as below?

```
variables, _ = model.init(jax.random.PRNGKey(0), x).pop('intermediates')
```

Reply:
Given #1284, `intermediates` will no longer be in the collection after init. I mistakenly updated the top example and missed the bottom one. Now both are the same.